src/v/cluster/tm_stm.cc

```diff
@@ -167,11 +167,53 @@ ss::future<checked<model::term_id, tm_stm::op_status>> tm_stm::do_barrier() {
         });
 }
 
-ss::future<> tm_stm::checkpoint_ongoing_txs() { return ss::now(); }
+ss::future<> tm_stm::checkpoint_ongoing_txs() {
+    if (!use_new_tx_version()) {
+        co_return;
+    }
+
+    auto can_transfer = [](const tm_transaction& tx) {
+        return !tx.transferring
+               && (tx.status == tm_transaction::ready || tx.status == tm_transaction::ongoing);
+    };
+    // Loop through all ongoing/pending txns in memory and checkpoint.
+    std::deque<tm_transaction> txes_to_checkpoint;
```

Comment:
why deque? seems like a vector would work (or a fragmented_vector<T> if the concern is contiguous memory allocation).

Reply:
> if the concern is contiguous memory allocation

Yea we didn't need contiguous allocation and just a quick insert and iteration. Do we prefer `fragmented_vector` in the code base? If so I can switch.

Reply:
yeh when you want a vector, but the contiguous allocation might be too big, fragmented_vector is the choice.

but no need to fix in this PR. you could slide it into some other PR